hm-module.nix

```diff
@@ -0,0 +1,49 @@
+self:
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+
+with lib;
+
+let
+  cfg = config.programs.krewfile;
+  finalPackage = self.packages.${pkgs.system}.krewfile;
+  krewfileContent = pkgs.writeText "krewfile" (concatStringsSep "\n" cfg.plugins);
+in
+{
+  options.programs.krewfile = {
+    enable = mkEnableOption "krewfile";
+
+    installKrew = mkEnableOption "installKrew";
+
+    krewPackage = mkOption {
+      type = types.package;
+      default = pkgs.krew;
+      defaultText = literalExpression "pkgs.krew";
+      description = ''
+        Krew package to install.
+        Only relevant if `programs.krewfile.installKrew` is enabled.
+      '';
+    };
+
+    plugins = mkOption {
+      type = with types; listOf str;
+      default = [ ];
+      description = "List of plugins to be written to the krewfile.";
+    };
+  };
+
+  config = mkIf cfg.enable {
+
+    home.packages = ([ finalPackage ] ++ optionals cfg.installKrew [ cfg.krewPackage ]);
+
+    home.activation.krew = hm.dag.entryAfter [ "writeBoundary" ] ''
+      run ${finalPackage}/bin/${finalPackage.pname} \
+        -command ${cfg.krewPackage}/bin/${cfg.krewPackage.pname} \
+        -file ${krewfileContent} -upgrade
```

Comment:
My current way is:
```
home.activation.krew = lib.hm.dag.entryAfter ["writeBoundary"] ''
    $DRY_RUN_CMD ${pkgs.krewfile}/bin/krewfile -command ${pkgs.krew}/bin/krew -file ${krewfile}
  '';
```

I think the `DRY_RUN_CMD` is something mentioned in the activation docs and I intentionally left out `-upgrade` to not upgrade on every rebuilt. Instead you can upgrade manually whenever needed. This is what the brew nix integration also does.
Ofc if you prefer it you could also add an option for that.

Reply:
See https://github.com/brumhard/krewfile/commit/67fb961afbd5c31b7bca8c9d6bb8bdce9681baf1

Reply:
`$DRY_RUN_CMD` is deprecated, so it's now simply `run`.
I added an enable `option` for upgrading, so the `-upgrade` is not called if configured.

Reply:
Ah didn't know 👍🏻